### ISSUE TITLE
Borders are not rendered properly

### ISSUE THREAD

**Issue by streetwolf** [Quicksaver/The-Fox--Only-Better#95]
I'm using the TFOB style Australis but this problem might apply to other styles as well.  As you can see in the screenshot the bookmarks border on the left and right consist of two borders.  I disabled CTR to rule out that add-on.  The double borders only appear if they are over a dark color like black in the screenshot.  I used cnn.com.
You might have to zoom the image to see the two borders.

![tfob](https://cloud.githubusercontent.com/assets/410799/6505410/f4131a3e-c30f-11e4-81e4-ce35af871171.jpg)

**Comment by Quicksaver** [comment]
There are a few specific situations (some of which aren't even that rare) where the australis style borders get screwed up in various forms. This is because of the fixes I had to put in place for #53...

So far, that bug isn't relevant in e10s, which is making me wait before I tackle this. Because the way the borders are implemented now is just kind of chaotic (and I kinda hate it), and if I can completely change them to simplify the process in e10s and make it easier to style those borders, I'd rather wait until then and focus on other stuff than on this right now.

I hope they're not a big bother in the meantime. :(

**Comment by Techy-Ben** [comment]
I've noticed the same problem, unfortunately. However, while waiting for Firefox to implement the border stye better, one temporary fix can be to use two toolbars in Slim Chrome, rather than one. See pictures below:

![slim chrome bottom border](https://cloud.githubusercontent.com/assets/6982633/8191585/dea44b32-1436-11e5-8294-30469fa9caec.png)
![slim chrome bottom border2](https://cloud.githubusercontent.com/assets/6982633/8191586/dea57b38-1436-11e5-9a5c-1296512c9fc4.png)

Note: I really like the look of Australis...if you have to change the styling in the future to fix the border problem, I hope you will keep the overall look of Australis. Please? :)

**Comment by Quicksaver** [comment]
> I've noticed the same problem

That's actually not the same problem, I'm guessing you're using some sort of theme that forces the navigation bar to be too short. Can you check if you have that same problem with the default theme? And if so, can you give me the name of that theme so I can check it out as well?

> if you have to change the styling in the future to fix the border problem, I hope you will keep the overall look of Australis.

Of course! What I meant was I don't like the way in which the style has to be implemented now (to avoid the issue in #53). When I change anything it will be the style's implementation (underlying code) and not the style itself, that will remain and it will still be Australis, if anything it will look sleeker. :)

**Comment by Techy-Ben** [comment]
Yes, I still have the problem with the default theme. It does not *look* as though the problem is there if the website I'm viewing has a white background, because then it's a white toolbar over a white background. However, if I scroll to a point where the toolbar is overlapping text or an image, I see this:

![slim chrome bottom border3](https://cloud.githubusercontent.com/assets/6982633/8197677/1874c2e2-1469-11e5-82c3-7b903b6cad57.png)
Moving everything from CTR's Additional Toolbar to the Bookmark Toolbar and disabling the Additional Toolbar did and affect the bottom border.

The theme I'm using is "Celebrate Americas Symbol." (With no apostrophe.)
As an aside, how do you access the code for a theme? Up to now, I've been tweaking my themes in userChrome.css. Are themes stored in the cloud for Firefox to load upon starting, or are they stored on my hard drive? I've found a few files and  a folder for the theme in my profile folder, but they *appear* to be placeholders more than actual code.

Great! I'm glad you don't plan on changing the style. :)

**Comment by Quicksaver** [comment]
> Yes, I still have the problem with the default theme.

Then it could be something, like a button or some text or placeholder, from another add-on that's pulling the navigation bar up; this does happen unfortunately and it's very hard to control. It could even be something from CTR but I've never seen it do that, maybe it's a specific widget it creates. Could you try disabling all your add-ons except for TFOB and see if that helps? It could narrow down the problem.

Also, if you have any navigation bar configurations in your userChrome.css it could also be interfering. Could you try clearing it out as well (make a backup of course so you can restore it later) when you disable the add-ons and see if that also helps?

> As an aside, how do you access the code for a theme?

The images for the currently used lightweight theme, like yours, are usually stored in the "lwtheme" folder in your profile. Other than that I don't really know unfortunately.

BTW it's almost definitely not the theme, lightweight themes like that one should be fully compatible already. They're little more than the background images after all.

**Comment by Techy-Ben** [comment]
Okay, I did a quick run through my add-ons, and Forcastfox (fix version) 2.2.6.1 is causing the glitch.

**Comment by Techy-Ben** [comment]
Specifically, the icon sets Abstract 1.0, Dashboard Weather 0.3a, Dashboard Weather 0.3b, and JyriK Shiny are causing the glitch.

I can move this (my specific bug that is) to the Forecastfox forum if you like, unless you want to patch it into TFOB.

**Comment by Quicksaver** [comment]
I'll take a look first, it's probably something easier to fix on my side anyway. Thank you for looking into the specifics, I know it must have been a bother to say the least.

**Comment by Quicksaver** [comment]
Unfortunately everything seems to show up correctly for me.

Could you get me a screenshot of the "Toolbar Display" section of your Forecastfox settings please, and also of the whole toolbars with the icons showing the problem if possible? Also, what operating system and firefox version are you using? And could you try the latest beta version 1.1.14b8 (https://github.com/Quicksaver/The-Fox--Only-Better/releases/tag/v1.1.14b8)? I doubt it will fix anything, but better make sure since that's what I'm testing with.

**Comment by Techy-Ben** [comment]
I took one screenshot for each of the problematic iconsets and also of one of the good iconset. Will these work?

![slim chrome bottom border and ff 1](https://cloud.githubusercontent.com/assets/6982633/8224639/eed5cb6c-1557-11e5-8493-2660d4a1b8b9.png)
![slim chrome bottom border and ff 2](https://cloud.githubusercontent.com/assets/6982633/8224640/eed900de-1557-11e5-8048-fed2725741c9.png)
![slim chrome bottom border and ff 3](https://cloud.githubusercontent.com/assets/6982633/8224643/eedb20e4-1557-11e5-8e10-896720a3aaa3.png)
![slim chrome bottom border and ff 4](https://cloud.githubusercontent.com/assets/6982633/8224642/eedb09e2-1557-11e5-9d95-25264abf392f.png)
![slim chrome bottom border and ff 5](https://cloud.githubusercontent.com/assets/6982633/8224641/eeda7aae-1557-11e5-9f82-3dd83155a52e.png)

I'm using Windows 8.1 and Firefox 38.0.5. I upgraded to TFOB 1.1.14b8, but as you suspected that did not change anything.

Don't worry, it wasn't a bother to look through the add-ons. For once I remembered to use the old 50% math rule, where I disabled all my add-ons, enabled 50%, and when the problem didn't pop up I enabled 50% of the remainder. The problem popped up then so I started disabling add-ons one at a time, starting with Forecastfox.

**Comment by Quicksaver** [comment]
Ah I see it now, I was trying Forecastfox in the navigation toolbar, but it works properly there already so that's why I wasn't seeing it before.

Would you please try beta version 1.1.14b9 (https://github.com/Quicksaver/The-Fox--Only-Better/releases/tag/v1.1.14b9) and see if it also fixes the issue for you?

**Comment by Techy-Ben** [comment]
Perfect! It looks great; thanks for taking the time to check it out and patch the fix in.

**Comment by Quicksaver** [comment]
You're very welcome, I'm glad it's looking good for you now. :)

Now to wait for e10s to get to be released so I can actually make those borders better...
